Feature Review: ADT Specification (#2)

Beatricenj opened the pull request:
This pull request contains changes to address Issue #2 by adding ADT specifications to transaction classes.

Closes #2

Beatricenj committed:
Update README.md

mulangonando commented:
@Beatricenj ,

For this too, you need to go through every code and add specifications for all ADT classes in your code.
Then commit the new resultant file into this branch.
I want to see actual codes written in the java files.

Beatricenj committed:
Added corrections.iml and updated .gitignore

Beatricenj committed:
Added corrections folder with updated files

Beatricenj commented:
Hello Sir,  I wanted to let you know that I have updated the pull request to include the `corrections` folder. The folder contains the corrections addressing Issues 1 and 2 

1. **Reverse Method Implementation**:
   - Added the `reverse()` method in `WithdrawalTransaction` to restore the original balance for reversible transactions.
   
2. **Incomplete Specification**:
   - Updated specifications with clear preconditions, postconditions, and behavior descriptions for the relevant methods and classes.

mulangonando commented on corrections/src/WithdrawalTransaction.java:
Hi @Beatricenj , 
Good job with the reverse() method.

This class is not fully specified though.
Please complete the specification of this class within this same Pull Request

mulangonando commented on corrections/src/TransactionInterface.java:
Hi @Beatricenj , 

Very good job: Specification looks good for this class. 
Always follow the following format for complete Specification:

/**
 * Name of function or class.
 *Description of what the function does
*
 * @param name_of_your_parameter - description of this parameter e.g. the type, range of values it can take and how it will be used in the function
 * @param name_of_your_parameter2 -  description of this parameter e.g. the type, range of values it can take and how it will be used in the function etc
 * @return type_to_returned e.g. float --- Description of the output e.g. what it will be used for, which function will need it etc
 * @throws SomeException describes what will cause this exception to be thrown. Note that throws is only added to the Specification if there is an exception thrown in the function. 
 * @see someOtherMethod()
 * @since 2.2.2 [Optional preferably if you intend to release versions of your software]
 * @author the_author [Optional --- Necesarry at the beginning of the file before the first class declaration]
 */
 
 Also take a look at this: https://docs.oracle.com/cd/E19683-01/806-7930/assert-13/index.html 
for more on how to specify pre and post-conditions as well as assertions

mulangonando commented on corrections/src/InsufficientFundsException.java:
@Beatricenj 

This class is not specified

mulangonando commented on corrections/src/DepositTransaction.java:
@Beatricenj 

This class is not Specified

mulangonando commented on corrections/src/BaseTransaction.java:
Hi @Beatricenj 

This class is not fully specified:

Always follow the following format for complete Specifications:

/**
 * Name of function or class.
 *Description of what the function does
*
 * @param name_of_your_parameter - description of this parameter e.g. the type, range of values it can take and how it will be used in the function
 * @param name_of_your_parameter2 -  description of this parameter e.g. the type, range of values it can take and how it will be used in the function etc
 * @return type_to_returned e.g. float --- Description of the output e.g. what it will be used for, which function will need it etc
 * @throws SomeException describes what will cause this exception to be thrown. Note that throws is only added to the Specification if there is an exception thrown in the function. 
 * @see someOtherMethod()
 * @since 2.2.2 [Optional preferably if you intend to release versions of your software]
 * @author the_author [Optional --- Necesarry at the beginning of the file before the first class declaration]
 */
 
 Also take a look at this: https://docs.oracle.com/cd/E19683-01/806-7930/assert-13/index.html 
for more on how to specify pre and post-conditions as well as assertions

mulangonando commented on corrections/src/BankAccount.java:
Hi @Beatricenj , 

This class is not fully specified.
Always follow the following format for complete Specifications:

/**
 * Name of function or class.
 *Description of what the function does
*
 * @param name_of_your_parameter - description of this parameter e.g. the type, range of values it can take and how it will be used in the function
 * @param name_of_your_parameter2 -  description of this parameter e.g. the type, range of values it can take and how it will be used in the function etc
 * @return type_to_returned e.g. float --- Description of the output e.g. what it will be used for, which function will need it etc
 * @throws SomeException describes what will cause this exception to be thrown. Note that throws is only added to the Specification if there is an exception thrown in the function. 
 * @see someOtherMethod()
 * @since 2.2.2 [Optional preferably if you intend to release versions of your software]
 * @author the_author [Optional --- Necesarry at the beginning of the file before the first class declaration]
 */
 
 Also, take a look at this: https://docs.oracle.com/cd/E19683-01/806-7930/assert-13/index.html 
for more on how to specify pre and post-conditions as well as assertions

mulangonando commented on corrections/.gitignore:
good file to have

mulangonando reviewed (changes requested):
Looks like you understand what is required but left a few things open-ended.
Please complete the tasks

Beatricenj committed:
Updated corrections folder with Main.java and other fixes

Beatricenj committed:
Updated src folder under corrections with the latest changes

Beatricenj committed:
Delete src directory

Beatricenj committed:
Delete WithdrawalTransaction.java

Beatricenj committed:
Delete DepositTransaction.java

Beatricenj committed:
Delete BaseTransaction.java

Beatricenj commented:
Hello sir,

I have updated the `corrections` folder in the `feature-review` branch.

Beatricenj commented:
Hello sir, I did all the corrections but did receive any feedback.

Beatricenj committed:
Update README.md

mulangonando reviewed (approved):
Looks Good
